src/core/tree_methods.test.ts

```diff
@@ -0,0 +1,77 @@
+import * as tree from './tree_methods'
+import * as _ from 'lodash'
+
+let root;
+
+let treeRoot = new tree.TreeNode("korzen", false);
```

Comment:
Ok, wtf is `false`? It doesn't make slightest sense for me while reading the tests -> an API redesign might be a good idea. Also: how about generic tree? After all the tree shouldn't care about it's contents. This will allow us to evolve domain (Resource types and associated methods) and nesting model independently.

Reply:
false is TreeNode object property. I will add tests to it.

Reply:
False hidden behind constructor default and will be removed.